app/src/main/java/com/sun_asterisk/moviedb_49/data/source/remote/fetchdata/GetMovieJsonFromUrl.kt

```diff
@@ -0,0 +1,36 @@
+package com.sun_asterisk.moviedb_49.data.source.remote.fetchdata
+
+import android.os.AsyncTask
+import com.sun_asterisk.moviedb_49.data.model.Movie
+import com.sun_asterisk.moviedb_49.data.source.remote.OnFetchDataJsonListener
+import java.lang.Exception
+
+class GetMovieJsonFromUrl(private var listener: OnFetchDataJsonListener<Movie>) :
+    AsyncTask<String, Unit, MutableList<Movie>>() {
+
+    private var mException: Exception? = null
+
+    override fun doInBackground(vararg string: String?): MutableList<Movie> {
+        var movies = mutableListOf<Movie>()
+        try {
+            val mParseJson = ParseUrlToJson()
+            val data = string[0]?.let { mParseJson.getJsonFromUrl(it) }
+            movies = data?.let { mParseJson.parseJsonToMovie(it) }!!
+        } catch (e: Exception) {
+            //listener.onFail(e)
+            mException = e
+        }
+        return movies
+    }
+
+    override fun onPostExecute(result: MutableList<Movie>?) {
+        super.onPostExecute(result)
+        if (mException == null) {
+            listener.onFail(mException!!)
+        } else {
+            if (result != null) {
+                listener.onSuccess(result)
+            }
+        }
```

Comment:
Không nên sử dụng dấu !! trong Kotlin, vì sử dụng !! nó sẽ bỏ qua việc check null, nếu giá trị null nó vẫn thows ra NullPointerExceptions --> Mất ý nghĩa của việc ra đời Null Safety 
```suggestion
        result?.let {
            callback.onSuccess(result)
        } ?: callback.onFailed(exception ?: Exception(MESSAGE_NULL_RESULT))
```
Nếu bạn lưu lại result thì làm tương tự, đổi chỗ exception và result